src/core/Sampler/Sampler.cpp

```diff
@@ -257,6 +280,30 @@ bool Sampler::renderNote( Note* pNote, unsigned nBufferSize, Song* pSong )
 		return 1;
 	}
 
+	// new instrument and note pan interaction--------------------------
+	// note pan moves the pan in a smaller pan range centered at instrument pan
+
+	// get the note and instrument pan parameters in [-1,1]
+	// TODO a new song member, set in preferences, must point the desired pan parameter Sampler member function
+	float (*getPan)( float, float );
+	// use "ratio" pan parameter (due to the current pan law)
+    getPan = &this->getRatioPan;
+	float fNotePan = getPan( pNote->get_pan_l(), pNote->get_pan_r() );
+	float fInstrPan = getPan( pInstr->get_pan_l(), pInstr->get_pan_r() );
+	
+	// resultant pan: note pan moves the pan in a smaller pan range centered at instrument pan
+	float fPan = fInstrPan + fNotePan * ( 1 - fabs( fInstrPan ) );
```

Comment:
Why do you use a weight for the note pan to make it more pronounced at the center?

Reply:
because the resultant note pan range extension is maximum (really from hard left to hard right) when the instrument pan is at center.
For example when instrument pan is hard panned to the right, there is 0 pan extension for the motion due to note_pan
that was the very meaning of "smaller pan range centered at..."
how small depends on the instrument_pan value.

Reply:
Like a "matrioska" pan
![image](https://user-images.githubusercontent.com/54219881/104101032-98183b00-529d-11eb-8de9-3f6dd3a11e05.png)